### PR TITLE
💫 Ability animations

### LINKED ISSUE #186: 💫 Add animation for abilities

### PR DESCRIPTION
**Pull request by Timsel1**
## Content

Abilities now have animations and the abilities trigger at the correct time during the animations.

### Added
`Assets/Animations/Player/AbilitySmite.anim` : Was created. Smite animation.
`Assets/Animations/Player/AbilitySpear.anim` : Was created. Spear throw animation, edited by me.
`Assets/Animations/Player/AbilitySpear_2.anim` : Was created. Original animation, kept in case it is needed later.
`Assets/Animations/Player/AbilityWave.anim` : Was created. Wave ability animation.

### Updated
`Assets/Animations/Player/Player.controller` : Was updated. Has the new animations and new conditions.
`Assets/Prefabs/Player/Player_1.prefab` : Was updated. Ability scripts can now be found on the player instead of the body, this way animation events can be used more effectively.
`Assets/Scripts/Player/Abilities/SmiteAbility.cs` : Was updated. Now triggers the animation and fixed bugs related to the new animations.
`Assets/Scripts/Player/Abilities/SpearAbility.cs` : Was updated. Now triggers the animation and fixed bugs related to the new animations.
`Assets/Scripts/Player/Abilities/WaveAbility.cs` : Was updated. Now triggers the animation and fixed bugs related to the new animations.

## Testing

The feature / Bugfix can be testing by following these steps:

FinalAbilityAnimation

Closes #186

### CODE REVIEW

File: Assets/Scripts/Player/Abilities/SpearAbility.cs [hunk 1]
```diff
@@ -9,13 +9,15 @@
 using System.Threading;
 using RogueApeStudio.Crusader.UI.Cooldown;
 using RogueApeStudio.Crusader.Audio;
+using log4net.Util;
```
Comment (NathanThus) [inline Assets/Scripts/Player/Abilities/SpearAbility.cs]: Why are you including this?

Comment (Timsel1) [reply Assets/Scripts/Player/Abilities/SpearAbility.cs]: not anymore, it was probably a quick action

File: Assets/Scripts/Player/Abilities/SpearAbility.cs
Comment (NathanThus) [inline Assets/Scripts/Player/Abilities/SpearAbility.cs:1]: Considering the entire system will need a rework in the near future, I won't nitpick on the style, but it's serizalized fields, then private fields.

**Review by NathanThus — APPROVED** [review]

File: Assets/Scripts/Player/Abilities/SpearAbility.cs
Comment (Timsel1) [reply Assets/Scripts/Player/Abilities/SpearAbility.cs:1]: Did 1 last push anyway to fix that for PlayerController, SpearAbility, WaveAbility and SmiteAbility

**Pull request merged**